LD2420 documentation outdated — examples use deprecated API that fails to compile on ESPHome 2025.x

### Feedback

The ESPHome documentation for the LD2420 still shows configuration options that were removed from the component in recent ESPHome versions (2025.x).
As a result, the official YAML examples fail to compile.

The docs still reference advanced LD2420 controls such as:

has_target

presence_timeout

min_gate_distance

max_gate_distance

gate_select

still_threshold

move_threshold

apply_config

factory_reset

restart_module

operating_mode

However, the current LD2420 component in ESPHome 2025.x does not include these fields.
Using the YAML exactly as documented results in compilation errors.

Steps to reproduce

Install ESPHome 2025.x (in Home Assistant or standalone)

Create device with esp32-c3-devkitm-1 board

Copy LD2420 configuration from the documentation into YAML

Attempt to compile

Build fails with the following error:

'class esphome::ld2420::LD2420Component' has no member named 'operating_selector_'
LOG_SELECT was not declared in this scope

Expected behavior

Documentation should reflect the current LD2420 component API

YAML examples should compile without modification

If advanced functions were removed, a breaking-change notice should be documented

Actual behavior

Documentation shows the old API

LD2420 component in ESPHome 2025.x supports only:

presence

moving_distance

stationary_distance

detection_distance

fw_version

All advanced controls referenced in docs are absent from the component

Compilation fails with missing-member errors

### URL

https://esphome.io/components/sensor/ld2420/

### Additional information

If what I say is true, I will help update the documentation.

No matter what I try in the yaml, I get:
src/esphome/components/ld2420/ld2420.cpp: In member function 'virtual void esphome::ld2420::LD2420Component::dump_config()':
src/esphome/components/ld2420/ld2420.cpp:209:44: error: 'class esphome::ld2420::LD2420Component' has no member named 'operating_selector_'
  209 |   LOG_SELECT("  ", "Operating Mode", this->operating_selector_);
      |                                            ^~~~~~~~~~~~~~~~~~~
src/esphome/components/ld2420/ld2420.cpp:209:3: error: 'LOG_SELECT' was not declared in this scope; did you mean 'LOG_SECRET'?
  209 |   LOG_SELECT("  ", "Operating Mode", this->operating_selector_);
      |   ^~~~~~~~~~
      |   LOG_SECRET
src/esphome/components/ld2420/ld2420.cpp: In member function 'virtual void esphome::ld2420::LD2420Component::setup()':
src/esphome/components/ld2420/ld2420.cpp:241:11: error: 'class esphome::ld2420::LD2420Component' has no member named 'operating_selector_'
  241 |     this->operating_selector_->publish_state(OP_SIMPLE_MODE_STRING);
      |           ^~~~~~~~~~~~~~~~~~~
src/esphome/components/ld2420/ld2420.cpp:246:11: error: 'class esphome::ld2420::LD2420Component' has no member named 'operating_selector_'
  246 |     this->operating_selector_->publish_state(OP_NORMAL_MODE_STRING);
      |           ^~~~~~~~~~~~~~~~~~~
src/esphome/components/ld2420/ld2420.cpp: In member function 'void esphome::ld2420::LD2420Component::set_operating_mode(const std::string&)':
src/esphome/components/ld2420/ld2420.cpp:387:11: error: 'class esphome::ld2420::LD2420Component' has no member named 'operating_selector_'
  387 |     this->operating_selector_->publish_state(state);
      |           ^~~~~~~~~~~~~~~~~~~
src/esphome/components/ld2420/ld2420.cpp:407:11: error: 'class esphome::ld2420::LD2420Component' has no member named 'operating_selector_'
  407 |     this->operating_selector_->publish_state(OP_SIMPLE_MODE_STRING);
      |           ^~~~~~~~~~~~~~~~~~~
*** [.pioenvs/waskotesp/src/esphome/components/ld2420/ld2420.cpp.o] Error 1
========================= [FAILED] Took 25.20 seconds =========================

--- a/esphome/components/ld2420/ld2420.cpp
+++ b/esphome/components/ld2420/ld2420.cpp
@@ -204,8 +204,10 @@ void LD2420Component::dump_config() {
   LOG_BUTTON("  ", "Factory Reset:", this->factory_reset_button_);
   LOG_BUTTON("  ", "Restart Module:", this->restart_module_button_);
 #endif
+#ifdef USE_SELECT
   ESP_LOGCONFIG(TAG, "Select:");
   LOG_SELECT("  ", "Operating Mode", this->operating_selector_);
+#endif
   if (ld2420::get_firmware_int(this->firmware_ver_) < CALIBRATE_VERSION_MIN) {
     ESP_LOGW(TAG, "Firmware version %s and older supports Simple Mode only", this->firmware_ver_);
   }
@@ -237,12 +239,20 @@ void LD2420Component::setup() {
   memcpy(&this->new_config, &this->current_config, sizeof(this->current_config));
   if (ld2420::get_firmware_int(this->firmware_ver_) < CALIBRATE_VERSION_MIN) {
     this->set_operating_mode(OP_SIMPLE_MODE_STRING);
-    this->operating_selector_->publish_state(OP_SIMPLE_MODE_STRING);
+#ifdef USE_SELECT
+    if (this->operating_selector_ != nullptr) {
+      this->operating_selector_->publish_state(OP_SIMPLE_MODE_STRING);
+    }
+#endif
     this->set_mode_(CMD_SYSTEM_MODE_SIMPLE);
     ESP_LOGW(TAG, "Firmware version %s and older supports Simple Mode only", this->firmware_ver_);
   } else {
     this->set_mode_(CMD_SYSTEM_MODE_ENERGY);
-    this->operating_selector_->publish_state(OP_NORMAL_MODE_STRING);
+#ifdef USE_SELECT
+    if (this->operating_selector_ != nullptr) {
+      this->operating_selector_->publish_state(OP_NORMAL_MODE_STRING);
+    }
+#endif
   }
 #ifdef USE_NUMBER
   this->init_gate_config_numbers();
@@ -382,8 +392,12 @@ void LD2420Component::set_operating_mode(const char *state) {
   // If unsupported firmware ignore mode select
   if (ld2420::get_firmware_int(firmware_ver_) >= CALIBRATE_VERSION_MIN) {
     this->current_operating_mode = find_uint8(OP_MODE_BY_STR, state);
-    // Entering Auto Calibrate we need to clear the privoiuos data collection
-    this->operating_selector_->publish_state(state);
+    // Entering Auto Calibrate we need to clear the previous data collection
+#ifdef USE_SELECT
+    if (this->operating_selector_ != nullptr) {
+      this->operating_selector_->publish_state(state);
+    }
+#endif
     if (current_operating_mode == OP_CALIBRATE_MODE) {
       this->set_calibration_(true);
       for (uint8_t gate = 0; gate < TOTAL_GATES; gate++) {
@@ -403,7 +417,11 @@ void LD2420Component::set_operating_mode(const char *state) {
     }
   } else {
     this->current_operating_mode = OP_SIMPLE_MODE;
-    this->operating_selector_->publish_state(OP_SIMPLE_MODE_STRING);
+#ifdef USE_SELECT
+    if (this->operating_selector_ != nullptr) {
+      this->operating_selector_->publish_state(OP_SIMPLE_MODE_STRING);
+    }
+#endif
   }
 }
 

